shared/logic/Update.ts

```diff
@@ -178,6 +178,14 @@ function tickTile(xy: Tile, gs: GameState, offline: boolean): void {
 
    Tick.next.totalValue += getBuildingValue(building);
 
+   forEach(building.resources, (res, amount) => {
+      if (amount <= 0) {
+         return;
+      }
+      Tick.next.totalValue += NoPrice[res] ? 0 : (Config.ResourcePrice[res] ?? 0) * amount;
+      safePush(Tick.next.resourcesByTile, res, xy);
```

Comment:
This is problematic. It will cause resources in buildings that are being upgraded used for other production.
I have pushed a fix, will close this PR

Reply:
Ah, not just a simple move of code then.